spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebFilter.java

```diff
@@ -156,6 +158,11 @@ TraceKeys traceKeys() {
 								addClassMethodTag(handlerMethod, span);
 								addClassNameTag(handlerMethod, span);
 							}
+							Object httpRoute = exchange
+									.getAttribute(HandlerMapping.BEST_MATCHING_PATTERN_ATTRIBUTE);
+							if (httpRoute != null) {
+								response.httpRoute = httpRoute.toString();
+							}
 							addResponseTagsForSpanWithoutParent(exchange, response, span);
 							handler().handleSend(response, t, span);
```

Comment:
I'd move the new DecoratedServerHttpResponse(exchange.getResponse()) here. it is only used by handleSend

Reply:
Valid point. Speaking of that - is there any reason for `request` and `response` being referenced from lambdas instead of retrieving them from the exchange? Not a big optimization but still...

Reply:
you can fix this so long as other tests still pass. We haven't optimized this code yet for performance.